resolver/src/test/java/io/netty/resolver/DefaultHostsFileEntriesResolverTest.java

```diff
@@ -19,16 +19,37 @@
  */
 package io.netty.resolver;
 
+import org.junit.Assert;
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
 
-import static org.junit.Assert.assertNotNull;
+import java.lang.reflect.Field;
+import java.net.InetAddress;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
 
 public class DefaultHostsFileEntriesResolverTest {
 
     @Test
-    public void testLocalhost() {
+    public void testResolutionIsCaseInsensitive() throws Exception {
         DefaultHostsFileEntriesResolver resolver = new DefaultHostsFileEntriesResolver();
-        assertNotNull("localhost doesn't resolve", resolver.address("localhost"));
-        assertNotNull("LOCALHOST doesn't resolve", resolver.address("LOCALHOST"));
+
+        Map<String, InetAddress> entriesMap = mock(Map.class);
+        Field entriesField = DefaultHostsFileEntriesResolver.class.getDeclaredField("entries");
+        entriesField.setAccessible(true);
+        entriesField.set(resolver, entriesMap);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        resolver.address("UPPERCASE");
+        resolver.address("uppercase");
```

Comment:
any reason for the host name change? consider reverting back to `LOCALHOST` and `localhost` ... awkward to read `uppercase` but its actually in lowercase.